### ISSUE TITLE
Add Getting Started drop down section within Project Home Page

### ISSUE THREAD

**Issue by ashlingam** [hackforla/website#237]
### Overview
Add Get Started drop down within Project-specific page 

### Action Items
Add Get Started drop down within Project-specific page, which contains Getting Started call-to-action with links to Markdown files. The projects that have it will uncomment it and add it in. If projects have a Getting Started guide, a Getting Started button will display. 

For iteration 0, every button will link to the generic Getting Started guide's README file. The README Getting Started guide, initially, will contain links to the project-specific Wikis until all projects have developed Getting Started guides.

### Resources/Instructions

**Comment by KianBadie** [comment]
Is this something we want to be implemented immediately, so it shows on the project homepage? Or is this going to be implemented along with the development of the project-specific page, and integrated to the home page when the project page get's rolled out? 

@ExperimentsInHonesty @harishlingam

**Comment by KianBadie** [comment]
From a Zoom meeting with Bonnie on 1/6:

- The main [Hack for LA Getting Started readme](https://github.com/hackforla/getting-started) will be on each project page under the "Getting Started" section.  
- Adding another field to each project md file called "resources" that is similar to the existing "links" field (which currently holds every link associated with the project). Resources will now contain every single link/resource associated with the project, meaning there will be overlap in displayed information. While the original "links" field are prioritized links. Meaning that these are the links that show towards the top of the project page in the "Links" section. These links on top are most likely seen first and are a shorter lists than the resources section. 
- Wikis/readmes of the project should show in the "Getting Started" section and the resources section. Meaning there will be overlap in displayed information. 
- In the "Getting Started Section", if the project already has a wiki/readme, it would show next to the hfla readme. If not, show a call for action to encourage users to help make a wiki/readme for the project. 
- Have wiki/readme show on main projects page as "Start Guide" link. However, the wording will be different on the project home page.

**Comment by KianBadie** [comment]
Creating a to do list for myself:

[X] Every project links to generic Hack for LA readme
[X] Check for existing readme or wiki and display it if exists
[X] If none exists, create call to action button to create one
[X] Create drop down menu feature, with the ability to close it

Also keeping track of things that came up during development that I wanted to bring up:

- What is going to be the name of the link of the readmes/wikis in the md card? Right now, I am just checking for links called "readme"
- Can a project have both a wiki and readme? If so, how would the naming of that work in the md card?
- Wording of links on dropdown menu

**Comment by ashlingam** [comment]
_Is this something we want to be implemented immediately, so it shows on the project homepage? Or is this going to be implemented along with the development of the project-specific page, and integrated to the home page when the project page get's rolled out?_

@KianBadie First, some clarification on terminology. We are no longer using "project-specific page":

Project Home Page: Page specific to each project
Projects List Page: Page that shows project cards for all projects with filters

In keeping with Agile development practice, my vote is to roll this out as soon as ready.

_What is going to be the name of the link of the readmes/wikis in the md card? Right now, I am just checking for links called "readme"_

How about we specify this to be called "project-onboarding"?

_Can a project have both a wiki and readme? If so, how would the naming of that work in the md card?_

I can imagine a scenario where a project has both (say if a team has isolated developer specific information to a README and everything else to a Wiki). Obviously, simpler if everything goes into a Wiki. Is this something we want to stipulate to all teams @ExperimentsInHonesty ?

**Comment by ashlingam** [comment]
I have asked @ysjiang18 to wireframe the drop down section so we can have clarity on what this should look like.

**Comment by ExperimentsInHonesty** [comment]
Q & A
Q1:  Can a project have both a wiki and readme? 
A1: If a project has a readme and a wiki, the wiki displays.  The wiki should link to the readme if there is one.

Q1a: If so, how would the naming of that work in the md card?
see the example of a md project card below.
md card example:
title: Shared Housing Project
description: The shared housing team works on creating an efficient & effective solution for matching multiple individuals who experience homelessness as potential co-tenants, and placing the matched individuals in suitable shared housing units.
image: /assets/images/projects/shared-housing-project.jpg
alt: "'linked arms'"
links:
  - name: Github
    url: 'https://github.com/hackforla/shared-housing'
  - name: Wiki
     url: https://github.com/hackforla/shared-housing/wiki
  - name: ReadMe
     url: https://github.com/hackforla/shared-housing/blob/master/README.md
 - name: Project Overview
    url: 
links_to_display: Github, Site
looking: Graphics Designer
location: Downtown LA
partner: 
status: Active

@harishlingam @KianBadie

**Comment by KianBadie** [comment]
Current progress:
![Screenshot from 2020-01-17 17-25-05](https://user-images.githubusercontent.com/18221058/72656178-6bfa0080-394e-11ea-8ac8-1cdab3430d35.png)
![Screenshot from 2020-01-17 17-24-44](https://user-images.githubusercontent.com/18221058/72656177-6b616a00-394e-11ea-814a-832a2bd4152d.png)

Feedback:
- A lot of red from a combination of the buttons and theme of the site. 
- Make the "x" gray
- Gray out getting started when the drop down is active. 
- Add a drop down icon next to the getting started instead of a clickable "Getting Started" button. This clears confusion of the button having any different functionality. The little "v" icon will be more clear you are initiating a drop down window. 
- Make the drop down in the window as the "Project Overview" section

This will be aimed to be done by next meeting 1/19

**Comment by KianBadie** [comment]
Current progress 1/25:
![dropdown](https://user-images.githubusercontent.com/18221058/73128948-2318fc00-3f8d-11ea-8d08-152483587dcd.gif)

Feedback:
- Having the project title instead of the book icon for project readme's would be better. As a result, we don't need to use "readme" as the title of the card, but can also call it "Getting Started"
- Move this section back to it's own card
- Instead of a red circle v indicating the drop down, just have a red down arrow
- Center text within the resource cards

This will be aimed to be done by next meeting 1/26

**Comment by ExperimentsInHonesty** [comment]
@KianBadie  please update with:
1. Progress
2. Blocks
3. Availability
4. ETA

**Comment by KianBadie** [comment]
Sorry for the late update @ExperimentsInHonesty, I am going to update my current progress now, but plan on having another update by our meeting on Sunday because I plan on working on it today and tomorrow.

1. Progress:
![Peek 2020-01-31 14-45](https://user-images.githubusercontent.com/18221058/73579818-6e715580-4438-11ea-9820-539628c25eee.gif)
I had a version where I made a mock readme for 311 data. But I recently reverted it back to normal. I will update what a version that uses a project with a readme soon. So please check back in later!

2. Blocks: Having an official format for what project.md files will follow would be nice. For example, what to call the 600x400/1500x700 image. 

3. Availability: I'm sorry, I'm not too sure what this means. If it is my availability, then I am trying to be free this Sunday for our 10:30 meeting. I got work scheduled but I found someone to cover it. I just need to solidify it.

4. ETA: I plan on having the page done by our meeting on Sunday the 2nd working with how the project cards are implemented up to this point.

**Comment by ExperimentsInHonesty** [comment]
What to do if they click the button "help us make one" If you are at hack night talk to the project manager, if you are remote, send us a slack message (slack message links to the team's slack channel).

**Comment by KianBadie** [comment]
### Lastly, some topics/points that have come up during my development that I wanted to post here for record keeping/possible discussion:

1. If project has no readme, we will display a call to action section. Once pressed, we talked about displaying a message that says to talk to a project manager if they are at a hack night or to send a slack message. So a link to the slack channel will be necessary. Not all projects have slack fields though. Can the call to action function work right now? Maybe instead display that the project doesn't have a readme but no call to action?
2. Right now, the project page just loops through the links section and looks for a field called "Readme." However, we have mentioned in a previous meetings that if a wiki exists it will display that instead. I need to update to the functionality to prioritize wikis before readmes.
3. It seems though that from the information we have been gathering for the project updates, readmes or wikis are not specifically being asked for. Is it going to be encouraged to make one or are wikis/readmes just going to be displayed it if they happen to have one?

### If more issues/topics come up I will be sure to raise them so they can be fixed immediately rather than dumped like this. My apologies for the load of text.

**Comment by ExperimentsInHonesty** [comment]
Answer 1: clicking the red "help make one" button will pop some text to the right that says - 
Once you are on the hack for LA slack channel (self invite), please post a note in our [channel] indicating your interest.  

the channel link will be delivered from the md file, because we will add a channel specific slack link to all project files.

**Comment by ExperimentsInHonesty** [comment]
Answer 2: Actually, lets display both but the order of display will be wiki, then readme.

**Comment by ExperimentsInHonesty** [comment]
Answer 3.  I think all projects have a readme at this point, but the red button will encourage any projects that don't.

**Comment by KianBadie** [comment]
Added a first draft of the pop up text for when a user is interested in helping create a readme:
![Peek 2020-02-08 22-43](https://user-images.githubusercontent.com/18221058/74097718-7712fd00-4ac4-11ea-964e-0bcd2ae9a6a1.gif)

**Comment by ExperimentsInHonesty** [comment]
1.  Changed some, still need to change to show wiki before readme
2. none
3. Sunday/monday
4. 2/10

**Comment by KianBadie** [comment]
@ExperimentsInHonesty 
Added logic that checks if a link field called "Wiki" exists:
![Screenshot from 2020-02-10 13-00-48](https://user-images.githubusercontent.com/18221058/74189557-8057b300-4c05-11ea-95aa-7ef584d89ca9.png)
The way the logic is implemented, it will search/display the Wiki first if it exists.

If either a Wiki or Readme does exist, then this call to action section will not show up. It will only show up if both the Wiki and Readme do not exists.
![Screenshot from 2020-02-10 13-03-05](https://user-images.githubusercontent.com/18221058/74189676-b4cb6f00-4c05-11ea-822f-a39398e6f066.png)


I tested this by adding a dummy link field to the Record Clearance project. Not sure what to do about the "profile pictures" of the Wiki and Readme, since they are both the same. Is that ok or something we would like to change?
